api/src/util/dict_util.py

```diff
@@ -40,3 +40,43 @@ def flatten_dict(in_dict: Any, separator: str = ".", prefix: str = "") -> dict:
 
     # value isn't a dictionary, so no more recursion
     return {prefix: in_dict}
+
+
+def diff_nested_dicts(dict1: dict, dict2: dict) -> list:
+    """
+        Compare two dictionaries (possibly nested), return a list of differences
+        with 'field', 'before', and 'after' for each key.
+
+
+        :param dict1 : The first dictionary.
+        :param dict2 : The second dictionary.
+        :return : Returns a list of dictionaries representing the differences.
+    a"""
+
+    flatt_dict1 = flatten_dict(dict1)
+    flatt_dict2 = flatten_dict(dict2)
+
+    diffs: list = []
+
+    all_keys = set(flatt_dict1.keys()).union(flatt_dict2.keys())  # Does not keep order
+
+    for k in all_keys:
+        values = [flatt_dict1.get(k, None), flatt_dict2.get(k, None)]
+        # convert values to set for comparison
+        v_a = _convert_to_set(values[0])
+        v_b = _convert_to_set(values[1])
```

Comment:
Sorry if when I quickly said this earlier, I'd still want to check if these are iterables before converting to a set (ie. if they're both integers, leave them as integers). Otherwise we have to make sure everything we ever feed through this is hashable into a set.

How about this - adjust/rename the `_convert_to_set` method to have an implementation like:
```py
if isinstance(data, (list, tuple)): # not iterable - see note
   return set(data)

return data
```
While the iterable check seems decent, you found that strings already have a problem that you have to build in an exception for. But bytes, dicts, strings, and a few other types end up needing special handling - it's probably easier to go without it.